bot/exts/recruitment/talentpool/_api.py

```diff
@@ -29,8 +29,8 @@ class Nomination(BaseModel):
 class NominationAPI:
     """Abstraction of site API interaction for talentpool."""
 
-    def __init__(self, site_api: APIClient):
-        self.site_api = site_api
+    def __init__(self, site_api_client: APIClient):
```

Comment:
IMO the name was clear before so renaming it is unnecessary and will only cause conflicts.

Reply:
How will it cause conflicts?

The reason I changed it is because when I was renaming the url value assigned to ''site_api'', this one got impacted as well and that's because it doesn't reflect exactly what it does since we clearly have different definitions of whay site_api is.
Plus, it's only a suffix I've added that reflects the class it's representing.

Reply:
I think this is a fine change, it improves clarity here. I don't think it's too big an issue either way.

Reply:
In general I would avoid renaming variables if not required because:
- It's not related to the PR (the name existed already), and slows down the PR if someone disagrees with the change.
- It causes merge conflicts with other PRs (I have one open I think).
- Variable names are personal preference, I prefer `site_api` which is why I used it when I wrote the code.

I don't think it improves clarity (`site_api.get` reads most fluently and clearly to me), so while I agree it's not a big issue, I don't think the change should be made.

Reply:
It's not really a matter of personal opinion to me personally, more of a semantics thing.

But sure, it wasn't intentional to get here, it's just that refactoring the original `site_api` changed its value as well, which made me think that not having two same names for different things is worth changing.

I've reverted it here 153ca4a1b31024a170222a18e7f1c7772c6d258f